testutil/base.go

```diff
@@ -62,8 +62,19 @@ func (s *BaseTest) AddCleanup(f func()) {
 	s.cleanupHandlers = append(s.cleanupHandlers, f)
 }
 
+// Backup a single element before further mocking.
+func Backup[T any](mockable *T) (restore func()) {
+	backup := *mockable
+
+	return func() {
+		*mockable = backup
+	}
+}
+
 // Backup the specified list of elements before further mocking.
-func Backup(mockablesByPtr ...interface{}) (restore func()) {
+func BackupMany(mockablesByPtr ...interface{}) (restore func()) {
+	// TODO find a type safe way of doing this when individual elements are
```

Comment:
There is no way with varadic functions and generics. You must use a slice of restore functions IMO.